macros/src/main/scala/shine/macros/Primitive.scala

```diff
@@ -0,0 +1,416 @@
+
+      val generated = q"""
+          ${if (visitAndRebuildMissing)
+        makeVisitAndRebuild(name, additionalParams, params)
+      else q""}
+
+          ${if (xmlPrinterMissing)
+        makeXMLPrinter(name, additionalParams, params)
+      else q""}
+         """
+
+      val expClass = (additionalParams match {
+        case List() =>
+          q"""
+          final case class $name(..$params) extends {} with ..$parents {
+            ..$body
+            ..$generated
+          }
+         """
+        case _ =>
+          val newParams = params map {
+            case ValDef(_, name, tpt, rhs) if rhs.isEmpty => q"val $name : $tpt"
+            case ValDef(_, name, tpt, rhs) => q"val $name : $tpt = $rhs"
+          }
+          q"""
+          final case class $name(..$additionalParams)
+                                (..$newParams) extends {} with ..$parents {
+            ..$body
+            ..$generated
+          }
+         """
+      }).asInstanceOf[ClassDef]
+
+      c.info(c.enclosingPosition,
+        s"generated `${name.toString}'\n$expClass", force = false)
+
+      expClass
+    }
+
+    def expPrimitivesFromClassDef: ClassDef => ClassDef = {
+      case q"case class $name(..$params) extends { ..$_ } with ..$parents {..$body} " =>
+        makeExpPrimitiveClass(
+          name.asInstanceOf[c.TypeName],
+          List(),
+          params.asInstanceOf[List[ValDef]],
+          body.asInstanceOf[List[Tree]],
+          parents.asInstanceOf[List[Tree]])
+      case q"final case class $name(..$params) extends { ..$_ } with ..$parents {..$body} " =>
+        makeExpPrimitiveClass(
+          name.asInstanceOf[c.TypeName],
+          List(),
+          params.asInstanceOf[List[ValDef]],
+          body.asInstanceOf[List[Tree]],
+          parents.asInstanceOf[List[Tree]])
+      case q"""case class $name(..$additionalParams)
+                               (..$params) extends { ..$_ } with ..$parents {..$body} """ =>
+        makeExpPrimitiveClass(
+          name.asInstanceOf[c.TypeName],
+          additionalParams.asInstanceOf[List[ValDef]],
+          params.asInstanceOf[List[ValDef]],
+          body.asInstanceOf[List[Tree]],
+          parents.asInstanceOf[List[Tree]])
+      case q"""final case class $name(..$additionalParams)
+                                     (..$params) extends { ..$_ } with ..$parents {..$body} """ =>
+        makeExpPrimitiveClass(
+          name.asInstanceOf[c.TypeName],
+          additionalParams.asInstanceOf[List[ValDef]],
+          params.asInstanceOf[List[ValDef]],
+          body.asInstanceOf[List[Tree]],
+          parents.asInstanceOf[List[Tree]])
+      case _ =>
+        c.abort(c.enclosingPosition, "expected a case class extends Primitive")
+    }
+
+    def makeAccPrimitiveClass(name: TypeName,
+                              additionalParams: List[ValDef],
+                              params: List[ValDef],
+                              body: List[Tree]): ClassDef = {
```

Comment:
Why does `makeExpPrimitiveClass` handle parents but this and `makeComPrimitiveClass` don't?